### DISCUSSION TITLE
SQL Server 2019

### DISCUSSION THREAD

**Discussion by vsudenko** [TPC-Council/HammerDB#143] [Q&A] [answered]
Hello,

I made performance tests for SQL server 2014 and 2019 on the same environment. I have good results for 2014 and low numbers for SQL server 2019. Are there special settings or this version is not supported yet?

**Comment by sm-shaw** [comment]
Hi, yes SQL Server 2019 has been tested for the next version, there are no special settings and it works with the current version. You need to provide much greater detail to help diagnose an issue. For example you don't say what workload you are running TPC-C or TPC-H?

**Comment by vsudenko** [comment]
Thanks for answer. Yes, I executed TPC-C test. Test setup:
HammerDB with 210 Warehouses, 21 Users (ensure “use all warehouse” is enabled).
Doing a TPM-C timed test, 2 minute ramp up, 5 minutes period.
Auto pilot test with scaling user for each test set to - 1,2,4,8,12,16,21
Test DB configured on E: drive of the server as synchronised DB with 210GB space and logs file size 112GB
For the constant test run during the disk switch will have 17 users and run for 40 minutes, swapping the disk 10 minutes into the test run.
![image](https://user-images.githubusercontent.com/36758749/90525943-17cb2f00-e170-11ea-8a47-8e66a1de49b0.png)

**Comment by sm-shaw** [comment, marked as answer]
The first reaction is that these numbers are relatively quite low - a simple test with SQL 2019 on an far from state of the art dev PC gives me 150134 NOPM from 345996 TPM. So the first step would be use SSMS Activity Monitor and identify the highest resource wait. IO for logging is often a bottleneck but clearly there is a dependency on the CPU resource as well. 
Also is this using the same version of HammerDB for both tests?- ie you are currently testing both database versions at the same time with the same stack? - if not what are the versions and are they both running on Windows with the same ODBC client library?

**Comment by sm-shaw** [comment]
Closed no further update
